src/main/java/com/linecorp/armeria/common/util/Version.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Retrieves the version information of available Armeria artifacts.
+ * <p>
+ * This class retrieves the version information from {@code META-INF/com.linecorp.armeria.versions.properties},
+ * which is generated in build time. Note that it may not be possible to retrieve the information completely,
+ * depending on your environment, such as the specified {@link ClassLoader}, the current
+ * {@link SecurityManager}.
+ * </p>
```

Comment:
A closing `<p>` tag causes Javadoc to fail - please remove it. http://stackoverflow.com/questions/5260368/which-tag-should-be-used-as-paragraph-separator-in-javadoc